net/graphsync_fetcher.go

```diff
@@ -101,26 +102,33 @@ func (gsf *GraphSyncFetcher) FetchTipSets(ctx context.Context, tsKey types.TipSe
 	return gsf.fetchRemainingTipsets(ctx, startingTipset, rpf, done)
 }
 
-func (gsf *GraphSyncFetcher) fetchFirstTipset(ctx context.Context, tsKey types.TipSetKey, rpf *requestPeerFinder) (types.TipSet, error) {
-	cids := tsKey.ToSlice()
+func (gsf *GraphSyncFetcher) fetchFirstTipset(ctx context.Context, key types.TipSetKey, rpf *requestPeerFinder) (types.TipSet, error) {
 	for {
-		err := gsf.fetchBlocks(ctx, cids, rpf.CurrentPeer())
-		if err == nil {
-			break
-		}
-		// something went wrong in a graphsync request, but we want to keep trying other peers, so
-		// just log error
-		logGraphsyncFetcher.Infof("Error occurred during Graphsync request: %s", err)
-		cids, err = gsf.missingCids(cids)
+		peer := rpf.CurrentPeer()
+		logGraphsyncFetcher.Infof("fetching initial tipset %s from peer %s", key, peer)
```

Comment:
This was intentional (with the expectation that graphsync internally would avoid actually fetching the already-stored blocks). However I've since reverted to the original logic (with better verification) as, coupled with the parallel requests, the change too hard to work with the existing unit tests.